torch/utils/data/datapipes/_hook_iterator.py

```diff
@@ -195,8 +195,13 @@ def wrap_generator(*args, **kwargs):
                     full_msg = f"{msg} {datapipe.__class__.__name__}({_generate_input_args_string(datapipe)})"
                     if len(e.args) == 0:  # If an exception message doesn't exist
                         e.args = (f'\nThis exception is {full_msg}',)
-                    elif msg not in e.args[0] and single_iterator_msg not in e.args[0]:
```

Comment:
IIUC the Error is raised due to `msg not in e.args[0]` where `e.args[0]` is an integer. Then why not we just do it:
```py
elif not isinstance(e.args[0], str):
    ...
elif msg not in e.args[0] and single_iterator_msg not in e.args[0]:
    ...
```

Reply:
Yea that's definitely better. Thaks!